File: LeetCode/Easy/1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
        String[] words = s.split(" ");
        int n = words.length;
        
        String[] sortedWords = new String[n];
        int[] indexes = new int[n];
        
        for (int i = 0; i < n; i++) {
            String word = words[i];
            indexes[i] = word.charAt(word.length() - 1) - '0';
            sortedWords[i] = word.substring(0, word.length() - 1);
        }
        
        // Bubble sort based on indexes
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (indexes[j] > indexes[j + 1]) {
                    int tempIndex = indexes[j];
                    indexes[j] = indexes[j + 1];
                    indexes[j + 1] = tempIndex;
                    
                    String tempWord = sortedWords[j];
                    sortedWords[j] = sortedWords[j + 1];
                    sortedWords[j + 1] = tempWord;
                }
            }
        }
        
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < n; i++) {
            result.append(sortedWords[i]);
            if (i != n - 1) {
                result.append(' ');
            }
        }
        
        return result.toString();
    }
}

File: LeetCode/Easy/1859-sorting-the-sentence/1859-sorting-the-sentence.cpp
struct Word {
    int index;
    string word;

    Word(int idx, const string& wd) : index(idx), word(wd) {}
};

bool compare(const Word& a, const Word& b) {
    return a.index < b.index;
}

class Solution {
public:
    string sortSentence(string s) {
        vector<string> words;
        size_t pos = 0, found;
        while ((found = s.find(' ', pos)) != string::npos) {
            words.push_back(s.substr(pos, found - pos));
            pos = found + 1;
        }
        words.push_back(s.substr(pos));

        vector<Word> pairs;
        for (const auto& word : words) {
            int lastIndex = word.back() - '0';
            pairs.emplace_back(lastIndex, word.substr(0, word.size() - 1));
        }

        sort(pairs.begin(), pairs.end(), compare);

        string result;
        for (size_t i = 0; i < pairs.size(); ++i) {
            result += pairs[i].word;
            if (i != pairs.size() - 1) result += ' ';
        }
        
        return result;
    }
};